src/main/java/uk/gov/hmcts/reform/sscs/service/DwpAddressLookupService.java

```diff
@@ -92,113 +91,88 @@ private Address buildAddress(OfficeAddress dwpAddress) {
-        return Optional.of(dwpMappings.getBereavementBenefit());
-    }
-
-    public Optional<OfficeMapping> iidbOfficeMapping(String dwpIssuingOffice) {
-        return getOfficeMappingByDwpIssuingOffice(dwpIssuingOffice, dwpMappings.getIidb());
-    }
-
-    public Optional<OfficeMapping> ucOfficeMapping(@SuppressWarnings("unused") String dwpIssuingOffice) {
-        return Optional.of(dwpMappings.getUc());
-    }
-
-    public Optional<OfficeMapping> esaOfficeMapping(String dwpIssuingOffice) {
-        return getOfficeMappingByDwpIssuingOffice(dwpIssuingOffice, dwpMappings.getEsa());
-    }
 
-    public Optional<OfficeMapping> pipOfficeMapping(String dwpIssuingOffice) {
-        String dwpIssuingOfficeStripped = ofNullable(substringBetween(dwpIssuingOffice,"(", ")"))
-                .orElse(dwpIssuingOffice.replaceAll("\\D+",""));
-
-        if (isEmpty(dwpIssuingOfficeStripped)) {
-            dwpIssuingOfficeStripped = dwpIssuingOffice;
+        //PIP search code
+        String dwpIssuingOfficeStripped = null;
+        if (findBenefitByShortName(benefitType).map(b -> b == PIP).orElse(false)) {
+            dwpIssuingOfficeStripped = ofNullable(substringBetween(dwpIssuingOffice, "(", ")"))
+                    .orElse(dwpIssuingOffice == null ? "" : dwpIssuingOffice.replaceAll("\\D+", ""));
         }
+        String dwpIssuingOfficeSearchKey = isEmpty(dwpIssuingOfficeStripped) ? dwpIssuingOffice : dwpIssuingOfficeStripped;
 
-        return getOfficeMappingByDwpIssuingOffice(dwpIssuingOfficeStripped, dwpMappings.getPip());
-    }
+        List<OfficeMapping> mappings = findBenefitByShortName(benefitType)
+                .map(b -> b.getOfficeMappings().apply(this)).orElse(List.of());
 
-    public Optional<OfficeMapping> getDefaultDwpMappingByBenefitType(String benefitType) {
-        return Benefit.findBenefitByShortName(benefitType)
-                .flatMap(benefit -> benefit.getDefaultOfficeMapping().apply(this));
+        if (mappings.size() > 1) {
+            return mappings.stream()
+                    .filter(office -> equalsAnyIgnoreCase(office.getCode(), stripToEmpty(dwpIssuingOfficeSearchKey)))
+                    .findFirst();
+        }
+        return mappings.stream().findFirst();
     }
 
-    public Optional<OfficeMapping> esaDefaultMapping() {
-        return of(dwpMappings.getEsa()).filter(OfficeMapping::isDefault).findFirst();
+    private String getDwpRegionalCenterByBenefitType(Optional<OfficeMapping> officeMapping) {
+        try {
+            if (officeMapping.map(m -> m.getMapping().getDwpRegionCentre() != null).orElse(false)) {
+                return officeMapping.map(mapping -> mapping.getMapping().getDwpRegionCentre()).orElse(null);
+            } else {
+                return officeMapping.map(mapping -> mapping.getMapping().getCcd()).orElse(null);
+            }
+        } catch (BenefitMappingException e) {
+            return null;
```

Comment:
you will get no 🍻 for this code.
This is api we control, so why is it fine to catch an Exception?

Reply:
This wasn't added in this PR. I believe this is to handle Case Loader calling this service with all ther other benefit types, including non mapped ones.